src/journal-remote/journal-upload.c

```diff
@@ -203,6 +239,22 @@ int start_upload(Uploader *u,
                         return log_oom();
                 h = l;
 
+                if (arg_compression != COMPRESSION_NONE) {
+                        _cleanup_free_ char *encoding = strdup(compression_to_string(arg_compression));
+                        if (!encoding)
+                                return log_oom();
+
+                        _cleanup_free_ char *header = NULL;
+                        header = strjoin("Content-Encoding: ", ascii_strlower(encoding));
```

Comment:
By making the `compression_table` lower, `ascii_strlower()` and strdup() can be dropped.